recover-chunk.py

```
@@ -0,0 +1,244 @@
+        nonce_i   = _gcm_derive_nonce(nonce0, block_index)
+        block_index += 1
+
+        if plain_len == 0:
+            # Sentinel — verify its GCM tag, then we're done
+            tag = f.read(GCM_TAG_SIZE)
+            if len(tag) < GCM_TAG_SIZE:
+                print("Error: truncated stream (missing sentinel tag)", file=sys.stderr)
+                sys.exit(1)
+            try:
+                aes.decrypt(nonce_i, tag, None)
+            except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)